torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -1868,7 +1915,8 @@ def create_from_config(cls, config: CppMicroGemmConfig):
 
     assert isinstance(n, int) or n.is_number, n
     assert isinstance(k, int) or k.is_number, k
-    m = V.graph.sizevars.size_hint(m, fallback=1) if isinstance(m, sympy.Expr) else m
+    dynamic_M = isinstance(m, sympy.Expr)
+    m = V.graph.sizevars.size_hint(m, fallback=1) if dynamic_M else m
     assert isinstance(m, int), m
```

Comment:
This causes failures of WOQ INT4 UT
```suggestion
    from ..utils import has_free_symbols
    dynamic_M = has_free_symbols((m,))
    m = V.graph.sizevars.size_hint(m, fallback=1) if dynamic_M else m
    assert isinstance(m, int) or m.is_number, m
```